dxr/plugins/clang/dxr-index.cpp

```diff
@@ -80,45 +79,57 @@ struct FileInfo {
   std::string realname;
   std::ostringstream info;
   bool interesting;
+  static std::string srcdir;  // the project source directory
```

Comment:
Why persist the strange decision of making these singletons? Conceptually, are they not one-per-DXRIndexAction-instantiation? Perhaps in a later PR, we could move them into IndexConsumer. We probably also have to put IndexConsumer in charge of creating FileInfos, since that's what's interested in most of those vars.